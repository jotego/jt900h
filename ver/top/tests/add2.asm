    ; ADD (mem),R
    main section code
    org 0
    ld a,0xbf    ; common header

    ld xix,data
    ; ADD (mem),R
    ld a,1
    add (xix),a
    ld w,(xix)
    cp w,0xff
    jp ne,bad_end
    or ra1,1

    ld a,0
    scf
    adc (xix+2),a
    ld w,(xix+2)
    cp w,0xf0
    jp ne,bad_end
    or ra1,2

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    end