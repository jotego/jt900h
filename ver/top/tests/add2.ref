0000f000 - 00000003 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000044 - 00000000 - 00000000 - 00000100
SR = f804
