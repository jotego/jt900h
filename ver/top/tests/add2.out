00010103 - 0000db00 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 0000cb00 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000002f - 00000000 - 00000000 - 00000000
